File: Backend/Newsletter/Data/SearchRequests/SearchRequestBase.cs
namespace Newsletter.Data.SearchRequests {
    public class SearchRequestBase {
        public string? SearchTerm { get; set; }
        public int Skip { get; set; } = 0;
        public int Take { get; set; } = 50;
        public string? OrderBy { get; set; }
        public bool Descending { get; set; } = false;
    }
}

File: Frontend/src/app/api/services/models/search-request-base.ts
import { ISearchRequestBase } from "./contracts/search-request-base.interface";

export class SearchRequestBase implements ISearchRequestBase {
    searchTerm?: string;
    skip: number = 0;
    take: number = 50;
    orderBy?: string;
    descending: boolean = false;

    constructor() {
    }

    public reset(): void {
        this.searchTerm = '';
        this.skip = 0;
        this.take = 50;
    }
}